utils/hiyobi.py

```diff
@@ -15,118 +14,117 @@ def __init__(self, artists, characters, groups, _id, parodys, tags, title):
         self.tags = tags
         self.title = title
 
+    @classmethod
+    def generator_hiyobi_info(cls, list_: list):
+        for info in list_:
+            yield cls(
+                info["artists"],
+                info["characters"],
+                info["groups"],
+                info["id"],
+                info["parodys"],
+                info["tags"],
+                info["title"],
+            )
+
+
+class HiyobiRequester:
+    @staticmethod
+    async def request(method, endpoint, json=None):
+        url = "https://api.hiyobi.me" + endpoint
+        async with aiohttp.ClientSession() as cs:
+            async with cs.request(method, url, json=json) as r:
+                if r.status == 404:
+                    return None
+                response = await r.json(content_type=None)
+            return response
 
-def generator_hiyobi_info(list_: list) -> HiyobiTagsModel:
-    for info in list_:
-        yield HiyobiTagsModel(
-            info["artists"],
-            info["characters"],
-            info["groups"],
-            info["id"],
-            info["parodys"],
-            info["tags"],
-            info["title"],
+    async def get_list(self, num):
+        response = await self.request("GET", f"/list/{num}")
+        return response
+
+    async def get_info(self, index) -> HiyobiTagsModel:
+        response = await self.request("GET", f"/gallery/{index}")
+        return HiyobiTagsModel(
+            response["artists"],
+            response["characters"],
+            response["groups"],
+            response["id"],
+            response["parodys"],
+            response["tags"],
+            response["title"],
         )
 
+    async def post_search(self, data):
```

Comment:
I'm not sure about the parameters of this part, I assumed it is `query`.

```suggestion
    async def post_search(self, data: str):
```